### PR TITLE
added erarchy links between models

### PR DESCRIPTION
**Pull request by aleksandr-hrankin**

### CODE REVIEW

File: src/main/resources/hibernate.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <mapping class="core.basesyntax.model.ma.Mentor"/>
+        <mapping class="core.basesyntax.model.ma.Person"/>
+        <mapping class="core.basesyntax.model.machine.Car"/>
+        <mapping class="core.basesyntax.model.machine.Machine"/>
+        <mapping class="core.basesyntax.model.machine.Truck"/>
+        <mapping class="core.basesyntax.model.zoo.Animal"/>
+        <mapping class="core.basesyntax.model.zoo.Cat"/>
+        <mapping class="core.basesyntax.model.zoo.Dog"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (KostiantynVorobiov) [inline src/main/resources/hibernate.cfg.xml:28]: add empty line

**Review by KostiantynVorobiov — APPROVED** [review]
Good job!

File: src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java [hunk 2]
```diff
@@ -12,11 +16,36 @@ public FigureDaoImpl(SessionFactory sessionFactory) {
...
+        }
     }
 
     @Override
     public List<T> findByColor(String color, Class<T> clazz) {
-        return null;
+        try (Session session = sessionFactory.openSession().getSession()) {
+            Query<T> findByColorQuery = session.createQuery(
+                    "from " + clazz.getSimpleName() + " f "
+                    + "where f.color = :color", clazz);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java:44]:
```suggestion
            Query<T> findByColorQuery = session.createQuery(
                    "from " + clazz.getSimpleName() + " f where f.color = :color", 
                    clazz);
```

File: src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java [hunk 3]
```diff
@@ -1,22 +1,50 @@
...
+            throw new DataProcessingException("Can't save machine " + machine, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java:40]: we should compare the machine year with the currentYear minus age

File: src/main/resources/hibernate.cfg.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        <property name="hibernate.dialect">org.hibernate.dialect.MySQL8Dialect</property>
+        <property name="hibernate.connection.driver_class">com.mysql.cj.jdbc.Driver</property>
+        <property name="hibernate.connection.username">root</property>
+        <property name="hibernate.connection.password">root</property>
+        <property name="hibernate.connection.url">jdbc:mysql://localhost:3306/sta?serverTimezone=UTC</property>
+        <property name="show_sql">true</property>
+        <property name="hibernate.hbm2ddl.auto">create-drop</property>
+
+        <mapping class="core.basesyntax.model.figure.Circle"/>
+        <mapping class="core.basesyntax.model.figure.Figure"/>
```
Comment (dmytro-hryhoruk) [inline src/main/resources/hibernate.cfg.xml:16]: do you need to map figure?

File: src/main/java/core/basesyntax/util/HibernateUtil.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+public class HibernateUtil {
+    private static SessionFactory sessionFactory;
+
+    public static SessionFactory getSessionFactory() {
+        if (sessionFactory == null) {
+            sessionFactory = new Configuration().configure().buildSessionFactory();
+        }
+        return sessionFactory;
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/util/HibernateUtil.java:14]: let's make SessionFactory as singleton

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**